File: matlab/getTransf.m
function [xform] = getTransf(num)
fname = sprintf('../../../hinterstoisser/OcclusionChallengeICCV2015/poses/Can/info_%05d.txt',num);
fid = fopen(fname);
tline = fgetl(fid);
i=0;
r= zeros(3,3);
t=zeros(3,1);
while ischar(tline)
    i=i+1;
    C = strsplit(tline,' ');
    
    if(i==5)
        r(1,1) = str2double(C(1));
        r(1,2) = str2double(C(2));
        r(1,3) = str2double(C(3));
    elseif(i==6)
            r(2,1) = str2double(C(1));
        r(2,2) = str2double(C(2));
        r(2,3) = str2double(C(3));
    elseif(i==7)
        r(3,1) = str2double(C(1));
        r(3,2) = str2double(C(2));
        r(3,3) = str2double(C(3));
    elseif(i==9)
        t(1,1) = str2double(C(1));
        t(2,1) = str2double(C(2));
        t(3,1) = str2double(C(3)); 
    end
    tline = fgetl(fid);  
end

xform = [r  t; 0 0 0 1];

File: python/getTransf.py
import numpy as np

def getTransf(num):
    fname = './poses/Can/info_%05d.txt'%(num);
    # file = open(fname, "r")
    # print(file)

    # tline = fgetl(fid);
    i = 0;
    r = np.zeros((3,3));
    t = np.zeros((3, 1));
    linesRead=0;
    # while ischar(tline)
    with open(fname) as file:
        for line in file:
            i = i + 1;
            C = line.split(" ")

            if (i == 5):
                # print(C)

                r[0,0] = float(C[0])
                r[0,1] = float(C[1])
                r[0,2] = float(C[2])
                linesRead=linesRead+1;
            elif (i == 6):
                # print(C)

                r[1, 0] = float(C[0])
                r[1, 1] = float(C[1])
                r[1, 2] = float(C[2])
                linesRead=linesRead+1;

            elif (i == 7):
                # print(C)

                r[2, 0] = float(C[0])
                r[2, 1] = float(C[1])
                r[2, 2] = float(C[2])
                linesRead=linesRead+1;

            elif (i ==9):
                # print(C)

                t[0, 0] = float(C[0]);
                t[1, 0] = float(C[1]);
                t[2, 0] = float(C[2]);
                linesRead=linesRead+1;



            xform = np.hstack((r ,  t))
            xform = np.vstack((xform,[0,0,0,1]))

    return xform,linesRead
# print(xform)